### PR TITLE
Remove BasicAer shot limit

### PR DESCRIPTION
**Pull request by alejomonbar**
### Summary
Fixes https://github.com/Qiskit/qiskit-terra/issues/7634
Executing the following code gives an error because `max_shots` for BasicAer has a limit of 65536 shots.

``` python
from qiskit import BasicAer, QuantumCircuit

backend = BasicAer.get_backend('qasm_simulator')
circuit = QuantumCircuit(1,1)
circuit.x(0)
circuit.measure(0,0)
job = backend.run(circuit, shots=int(1e6))
```

### Details and comments

The idea is that BasicAer does not have a limit in terms of the number of shots. Instead, the default value equals zero which works as no restriction on the number of shots.

### CODE REVIEW

**Comment by alejomonbar** [comment]
@kevinsung now this issue is passing.

**Comment by kevinsung** [comment]
@alejomonbar Please use descriptive titles for your pull requests. For example, "Remove BasicAer shot limit"

**Review by jakelishman — COMMENTED** [review]
Looks good to me, thanks!  Please can you also add an "upgrade" release note that says that the limit has been removed?

Do you know how long the new test takes to run?

**Comment by alejomonbar** [comment]
> Looks good to me, thanks! Please can you also add an "upgrade" release note that says that the limit has been removed?
> 
> Do you know how long the new test takes to run?

Thank you @jakelishman, I added the release note. About the time, I don't know if there is any command to know it, but, the circuit for the test is simple and only requires 2e6 shots, I don't think this adds much extra time for the tox test.

File: releasenotes/notes/remove-basicaer-shot-limit-b7cd4e6f6739885c.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+upgrade:
+  - |
+    The maximum number of shots in BasicAer backend has been removed. Now, by default, it has been set to zero which is the special case for no evaluation of a maximum number of shots.
```
[the remark this answers is not in the document]
Comment (alejomonbar) [reply releasenotes/notes/remove-basicaer-shot-limit-b7cd4e6f6739885c.yaml:4]: Thank you @kevinsung.

**Review by jakelishman — APPROVED** [review]
Two million shots is actually quite a lot, especially for a simulator implemented in pure Python.  As things stood, the test took about 8 seconds.  Taken alone, that's ok, but Terra's test suite has something like ten to fifteen thousand tests in it, and at that scale, 8 seconds is a long time to test on a simple toggle.

While writing the test was definitely the right thing to do, in this very particular circumstance, I think the best course of action is just to not have one.  It can be hard to test for the absence of a limit efficiently, and this particular feature is something that's very unlikely to be accidentally changed - if a limit is re-instated in the future, it'll almost certainly be somebody having made a deliberate technical choice.

Thanks for the contribution!

**Pull request merged**